client/my-sites/plans-features-main/index.jsx

```diff
@@ -129,12 +147,33 @@ export class PlansFeaturesMain extends Component {
 			redirectTo,
 			siteId,
 			plansWithScroll,
-			translate,
 			customHeader,
 		} = this.props;
 
 		const plans = this.getPlansForPlanFeatures();
 		const visiblePlans = this.getVisiblePlansForPlanFeatures( plans );
+		const availablePlans = this.isDisplayingPlansNeededForFeature()
+			? visiblePlans.filter( plan => {
+					if ( isEcommercePlan( selectedPlan ) ) {
+						return isEcommercePlan( plan );
+					}
+					if ( isBusinessPlan( selectedPlan ) ) {
+						return isBusinessPlan( plan ) || isEcommercePlan( plan );
+					}
+					if ( isPremiumPlan( selectedPlan ) ) {
+						return isPremiumPlan( plan ) || isBusinessPlan( plan ) || isEcommercePlan( plan );
+					}
+
+					if ( isPersonalPlan( selectedPlan ) ) {
```

Comment:
One of the reasons why we show the tabbed navigation that separates plans for personal sites from plans for business sites is for not showing more than 3 plans at time.

Also, given that the Personal plan is the "lowest" one, can we maybe don't filter the plans on that case?

I guess we can add one more check to `isDisplayingPlansNeededForFeature`:

```
if (
	isValidFeatureKey( selectedFeature ) &&
	getPlan( selectedPlan ) &&
	! isPersonalPlan( selectedPlan ) && 
	! previousRoute.startsWith( '/plans/' )
) {
	return true;
}
```

That way we can remove the 4 plans returned here.

Reply:
See for example what happens currently with the upsell nudge displayed on the activity log. That feature is included in the Personal plan, so the plans displayed after clicking on "View all plans" don't change at all.

![Apr-17-2020 13-02-29](https://user-images.githubusercontent.com/1233880/79562800-e18b7f00-80ab-11ea-8f97-9f1b79f536c9.gif)

Reply:
That makes sense! Updated the PR with that change. :)